src/cdi/CPCommon.hh

```diff
@@ -41,6 +44,16 @@ private:
 
 public:
   int32_t get_zaid() const { return zaid; }
+  int32_t get_z() const {
+    if (zaid != -1) {
+      // Integer division to extract the ZZZ from ZAID format ZZZAAA
+      return zaid / 1000;
+    } else {
+      // Electrons have special ZAID = -1
+      return -1;
```

Comment:
Can you add a unit test that captures this branch?

Reply:
Done (test for new TR stopping power model tests both ion and electron cases, which differ)